pkg/services/cloudmigration/cloudmigrationimpl/xorm_store.go

```diff
@@ -433,3 +448,45 @@ func (ss *sqlStore) decryptToken(ctx context.Context, cm *cloudmigration.CloudMi
 
 	return nil
 }
+
+func (ss *sqlStore) DeleteMigrationSessionWithRelatedElements(ctx context.Context, sessionUID string) (*cloudmigration.CloudMigrationSession, error) {
+	// first we try to delete all the associated information to the session
+	q := cloudmigration.ListSnapshotsQuery{
+		SessionUID: sessionUID,
+		Page:       1,
+		// passing -1 will return all the elements
+		Limit: -1,
+	}
+	snapshots, err := ss.GetSnapshotList(ctx, q)
+	if err != nil {
+		return nil, fmt.Errorf("getting migration snapshots from db: %w", err)
+	}
+	for _, snapshot := range snapshots {
+		err := ss.DeleteSnapshotResources(ctx, snapshot.UID)
+		if err != nil {
+			return nil, fmt.Errorf("deleting snapshot resource from db: %w", err)
+		}
+		err = ss.DeleteSnapshot(ctx, snapshot.UID)
+		if err != nil {
+			return nil, fmt.Errorf("deleting snapshot from db: %w", err)
+		}
+
+		err = deleteLocalFiles(snapshot)
```

Comment:
I don't think the store should be responsible for deleting the local files, that should be done by the caller after the fact. It might be slightly weird, but my workaround would be to just add the list of localfiles to this method's return values.